src/type_checker/checker.rs

```diff
@@ -389,6 +389,112 @@ impl<B: Backend> TypeChecker<B> {
                     | Op2::Division
                     | Op2::BoolAnd
                     | Op2::BoolOr => lhs_node.typ,
+                    Op2::PlusEqual => {
+                        // todo: better way to do this instead of copying assignment logic?
```

Comment:
Here, I copied the same logic from Expr::Assignment. I am not sure if this is the correct way to do it.